tests/robot-tests/tests/admin/analyst/role_ui_permissions/release_and_publication_role_end_to_end_permissions.robot

```diff
@@ -41,7 +41,7 @@ Check publication owner can update methodology summary
 
 Check publication owner cannot approve methodology for publication
     user clicks link    Sign off
-    user waits until page does not contain loading spinner
+    user waits until page finishes loading
```

Comment:
Makes me wonder if moving forwards with this, if we should building this in as a compulsory step before any step, or before certain subsets of our steps.

Or flip it and include it *after* any of our steps that expect the page to update based upon an action e.g. all "user clicks link/button", "user goes to url", etc etc.

Not suggesting we do this as part of this PR, but I'd say that I'm seeing this step appearing very regularly right after certain other steps have been called.

To start this, we could look at everywhere where we have used this step `user waits until page finishes loading` and see what the preceding steps are.  If we see enough repetition with certain steps, we can build it into them (and maybe provide an optional way to suppress waiting for the page to finish loading if we really don't want to wait).

We might be concerned that we could pay a performance penalty for this, but I'd say that that's easily tested with a before-and-after set of timings.

Reply:
Cool no worries, thanks for checking it out.